File: framework/src/main/java/com/framework/base/SuperViewHolder.java
package com.framework.base;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.util.SparseArray;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

/**
 * by y on 2016/9/16.
 */
public class SuperViewHolder extends RecyclerView.ViewHolder {

    private SparseArray<View> viewSparseArray;
    private Context context;

    public SuperViewHolder(final View itemView) {
        super(itemView);
        viewSparseArray = new SparseArray<>();
        context = itemView.getContext();
        itemView.setTag(viewSparseArray);
    }

    public <T extends View> T get(int id) {
        View childView = viewSparseArray.get(id);
        if (childView == null) {
            childView = itemView.findViewById(id);
            viewSparseArray.put(id, childView);
        }
        return (T) childView;
    }

    public Context getContext() {
        return context;
    }

    public TextView getTextView(int id) {
        return get(id);
    }

    public ImageView getImageView(int id) {
        return get(id);
    }

    public void setTextView(int id, CharSequence charSequence) {
        getTextView(id).setText(charSequence);
    }

    public void setTextColor(int id, int color) {
        getTextView(id).setTextColor(color);
    }

}

File: framework/src/main/java/com/framework/base/SuperViewHolder.kt
package com.framework.base

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.util.SparseArray
import android.view.View
import android.widget.ImageView
import android.widget.TextView

/**
 * by y on 2016/9/16.
 */
class SuperViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    private val viewSparseArray: SparseArray<View> = SparseArray()
    val context: Context = itemView.context

    init {
        itemView.tag = viewSparseArray
    }

    operator fun <T : View> get(id: Int): T {
        var childView: View? = viewSparseArray.get(id)
        if (childView == null) {
            childView = itemView.findViewById(id)
            viewSparseArray.put(id, childView)
        }

        return (childView as T?)!!
    }

    fun getTextView(id: Int): TextView {
        return get(id)
    }

    fun getImageView(id: Int): ImageView {
        return get(id)
    }

    fun setTextView(id: Int, charSequence: CharSequence) {
        getTextView(id).text = charSequence
    }

    fun setTextColor(id: Int, color: Int) {
        getTextView(id).setTextColor(color)
    }

}
